clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1376,7 +1376,7 @@ public void handleResponse(AbstractResponse response) {
                     fatalError(error.exception());
                     return;
                 } else if (error == Errors.INVALID_TXN_STATE) {
-                    fatalError(new KafkaException(error.exception()));
```

Comment:
As mentioned, in the other instance, this probably won't be returned to the client, but in the case that it was -- could there be a client that expects the wrapped error? At this point, it's probably not an issue, but just curious the implication.